File: puremvc/puremvc/MVC/Patterns/Observer/Notification.cs
using MVC.Interfaces;

namespace MVC.Patterns.Observer
{
    public class Notification: INotification
    {
        public Notification(string name, object body = null, string type = null)
        {
            Name = name;
            Body = body;
            Type = type;
        }

        public override string ToString()
        {
            var msg = "Notification Name:" + Name;
            msg += "\nBody: " + ((Body == null) ? "null" : Body.ToString());
            msg += "\nType: " + ((Type == null) ? "null" : Type);
            return msg;
        }

        public string Name{get;}
        public object Body{get; set;}
        public string Type{get; set;}
    }
}

File: framework/project/base/XScripts/Frameworks/MVC/Patterns/Observer/Notification.lua
local Notification = Class("Notification", {})

function Notification:Ctor(name, body, type)
    self.name = name
    self.body = body
    self.type = type
end

function Notification:GetName()
    return self.name
end
function Notification:SetBody(body)
    self.body = body
end
function Notification:GetBody()
    return self.body
end
function Notification:SetType(type)
    self.type = type
end
function Notification:GetType()
    return self.type
end
function Notification:ToString()
    local msg = "Notification Name:" .. self:GetName()
    msg = msg .. "\nBody:" .. tostring(self:GetBody())
    msg = msg .. "\nType:" .. self:GetType()
    return msg
end

return Notification
